test/extended/router/http2.go

```diff
@@ -172,25 +255,123 @@ var _ = g.Describe("[sig-network-edge][Conformance][Area:Networking][Feature:Rou
 						return false, nil // could be 503 if service not ready
 					}
 					if tc.statusCode == 0 {
+						resp.Body.Close()
 						return false, nil
 					}
 					if resp.StatusCode != tc.statusCode {
+						resp.Body.Close()
 						e2e.Logf("[test #%d/%d]: config: %s, expected status: %v, actual status: %v", i+1, len(testCases), testConfig, tc.statusCode, resp.StatusCode)
+						return false, nil
 					}
-					return resp.StatusCode == tc.statusCode, nil
+					return true, nil
 				})).NotTo(o.HaveOccurred())
 
 				if tc.expectedGetError != "" {
 					continue
 				}
 
+				o.Expect(resp).ToNot(o.BeNil(), testConfig)
 				o.Expect(resp.StatusCode).To(o.Equal(tc.statusCode), testConfig)
 				o.Expect(resp.Proto).To(o.Equal(tc.frontendProto), testConfig)
 				body, err := ioutil.ReadAll(resp.Body)
 				o.Expect(err).NotTo(o.HaveOccurred(), testConfig)
+				resp.Body.Close()
 				o.Expect(string(body)).To(o.Equal(tc.backendProto), testConfig)
 				o.Expect(resp.Body.Close()).NotTo(o.HaveOccurred())
 			}
 		})
 	})
 })
+
+func getDefaultIngressClusterDomainName(oc *exutil.CLI, timeout time.Duration) (string, error) {
+	var domain string
+
+	if err := wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		ingress, err := oc.AdminConfigClient().ConfigV1().Ingresses().Get(context.TODO(), "cluster", metav1.GetOptions{})
+		if err != nil {
+			e2e.Logf("Get ingresses.config/cluster failed: %v, retrying...", err)
+			return false, nil
+		}
+		domain = ingress.Spec.Domain
+		return true, nil
+	}); err != nil {
+		return "", err
+	}
+
+	return domain, nil
+}
+
+func getRouterService(oc *exutil.CLI, timeout time.Duration, name string) (*v1.Service, error) {
+	var svc *v1.Service
+
+	if err := wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		o, err := oc.AdminKubeClient().CoreV1().Services("openshift-ingress").Get(context.Background(), name, metav1.GetOptions{})
+		if err != nil {
+			e2e.Logf("Get router service %s failed: %v, retrying...", name, err)
+			return false, nil
+		}
+		svc = o
+		return true, nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return svc, nil
+}
+
+func getCanaryImage(oc *exutil.CLI) (string, error) {
+	o, err := oc.AdminConfigClient().ConfigV1().ClusterOperators().Get(context.Background(), "ingress", metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	for _, v := range o.Status.Versions {
+		if v.Name == "canary-server" {
```

Comment:
There is a `const` you can use, `CanaryImageVersionName` in pkg/operator/controller/status/controller.go `status` package.

Reply:
Probably not worth vendoring the ingress-operator for this.

Reply:
I didn't want to vendor for a single string constant (given that it is unlikely to change).